client/src/main/java/org/eclipse/hono/client/impl/RegistrationClientImpl.java

```diff
@@ -171,8 +192,36 @@ public void get(final String deviceId, final Handler<AsyncResult<RegistrationRes
      */
     @Override
     public void assertRegistration(final String deviceId, final Handler<AsyncResult<RegistrationResult>> resultHandler) {
+
         Objects.requireNonNull(deviceId);
         Objects.requireNonNull(resultHandler);
-        createAndSendRequest(ACTION_ASSERT, createDeviceIdProperties(deviceId), null, resultHandler);
+
+        final String key = "assert-" + deviceId;
```

Comment:
The "assert-" prefix looks like a convention regarding the name patterns for cache entries (currently only "assert-" exists as prefix, but maybe we will have more regarding devices in the future?).
If that is the intention, it would be good to place this to a helper method (which can be used in the test classes then as well).
WDYT?

Reply:
Since the caches are client internal only, and no external classes need to know about the naming pattern used for the keys, I do not think that we need to expose them in a helper method. I consider the unit test class as part of the implementation so I do not think that there is a problem with the fact that the unit tests check for the same string literals.